lib/generators/hobo/migration/migrator.rb

```diff
@@ -393,22 +410,33 @@ def change_table(model, current_table_name)
         end
 
         def change_indexes(model, old_table_name)
-          return [[],[]] if Migrator.disable_constraints || model.is_a?(HabtmModelShim)
+          return [[],[]] if Migrator.disable_constraints
           new_table_name = model.table_name
           existing_indexes = HoboFields::Model::IndexSpec.for_model(model, old_table_name)
-          model_indexes = model.index_specs
-          add_indexes = model_indexes - existing_indexes
-          drop_indexes = existing_indexes - model_indexes
+          model_indexes_with_equivalents = model.index_specs_with_primary_key
+          model_indexes = model_indexes_with_equivalents.map do |i|
+            if i.explicit_name.nil?
+              if ex = existing_indexes.find { |e| i != e && e.equivalent?(i) }
+                i.with_name(ex.name)
+              end
+            end || i
+          end
+          existing_has_primary_key = existing_indexes.any? { |i| i.name == 'PRIMARY_KEY' }
+          model_has_primary_key    = model_indexes.any?    { |i| i.name == 'PRIMARY_KEY' }
+
+          add_indexes_init = model_indexes - existing_indexes
+          drop_indexes_init = existing_indexes - model_indexes
           undo_add_indexes = []
           undo_drop_indexes = []
-          add_indexes.map! do |i|
-            undo_add_indexes << drop_index(old_table_name, i.name)
-            i.to_add_statement(new_table_name)
-          end
-          drop_indexes.map! do |i|
-            undo_drop_indexes << i.to_add_statement(old_table_name)
-            drop_index(new_table_name, i.name)
+          add_indexes = add_indexes_init.map do |i|
+            undo_add_indexes << drop_index(old_table_name, i.name) unless i.name == "PRIMARY_KEY"
+            i.to_add_statement(new_table_name, existing_has_primary_key)
           end
+          drop_indexes = drop_indexes_init.map do |i|
+            undo_drop_indexes << i.to_add_statement(old_table_name, model_has_primary_key)
+            drop_index(new_table_name, i.name) unless i.name == "PRIMARY_KEY"
+          end.compact
+
```

Comment:
I believe this will not handle changes to the primary key.   This doesn' t matter with rails where the primary key is id and an int.   Did you have to do something to get the oddball warehouse primary keys to work?